src/System.Management.Automation/DscSupport/MofCimDSCParser.cs

```diff
@@ -0,0 +1,478 @@
+
+                powerShell.AddScript(script);
+                powerShell.AddArgument(targetType);
+                powerShell.AddArgument(moduleName);
+
+                Collection<PSObject> psExecutionResult = powerShell.Invoke();
+                if (psExecutionResult.Count == 1)
+                {
+                    targetObject = psExecutionResult[0].BaseObject;
+                }
+                else
+                {
+                    Exception innerException = null;
+                    if (powerShell.Streams.Error != null && powerShell.Streams.Error.Count > 0)
+                    {
+                        innerException = powerShell.Streams.Error[0].Exception;
+                    }
+
+                    errorMessage = string.Format(CultureInfo.CurrentCulture, ParserStrings.InstantiatePSClassObjectFailed, className);
+                    var invalidOperationException = new InvalidOperationException(errorMessage, innerException);
+                    throw invalidOperationException;
+                }
+            }
+
+            foreach (var property in instance.CimInstanceProperties)
+            {
+                if (property.Value != null)
+                {
+                    MemberInfo[] memberInfo = targetType.GetMember(property.Name, BindingFlags.Public | BindingFlags.Instance);
+
+                    // verify property exists in corresponding class type
+                    if (memberInfo == null || memberInfo.Length > 1 || !(memberInfo[0] is PropertyInfo || memberInfo[0] is FieldInfo))
+                    {
+                        errorMessage = string.Format(CultureInfo.CurrentCulture, ParserStrings.PropertyNotDeclaredInPSClass, new object[] { property.Name, className });
+                        var invalidOperationException = new InvalidOperationException(errorMessage);
+                        throw invalidOperationException;
+                    }
+
+                    var member = memberInfo[0];
+                    var memberType = (member is FieldInfo)
+                        ? ((FieldInfo)member).FieldType
+                        : ((PropertyInfo)member).PropertyType;
+
+                    object targetValue = null;
+                    switch (property.CimType)
+                    {
+                        case Microsoft.Management.Infrastructure.CimType.Instance:
+                            {
+                                var cimPropertyInstance = property.Value as CimInstance;
+                                if (cimPropertyInstance != null &&
+                                    cimPropertyInstance.CimClass != null &&
+                                    cimPropertyInstance.CimClass.CimSystemProperties != null &&
+                                    string.Equals(
+                                        cimPropertyInstance.CimClass.CimSystemProperties.ClassName,
+                                        "MSFT_Credential", StringComparison.OrdinalIgnoreCase))
```

Comment:
```suggestion
                                if (string.Equals(
                                    (property.Value as CimInstance)?.CimClass?.CimSystemProperties,
                                    "MSFT_Credential",
                                    StringComparison.OrdinalIgnoreCase))
```

Reply:
MofCimDSCParser.cs is existing code (that was moved to a new file, so it shows as green in the diff) that I'd like to avoid changing unless absolutely necessary.